homeassistant/helpers/config_validation.py

```diff
@@ -65,9 +68,36 @@ def boolean(value: Any) -> bool:
     return bool(value)
 
 
-def isfile(value):
+def isfile(value: Any) -> str:
     """Validate that the value is an existing file."""
-    return vol.IsFile('not a file')(value)
+    if value is None:
+        raise vol.Invalid('None is not file')
+    return vol.IsFile('not a file')(str(value))
+
+
+def file(value: Any) -> str:
```

Comment:
Today, the function `isfile` checks if it exists (os.path.isfile)

The name `file` is not that clear, maybe we can call it `filename` / `isfilename` to test if it is a valid filename (or writable path)?